### PR TITLE
docs: add floating Table of Contents

### PR DESCRIPTION
**Pull request by andreialecu**
**What's the problem this PR addresses?**

Adds a floating, dynamic, table of contents to the side of articles.

**How did you fix it?**

Created a React component that was adapted from a blog article at https://janosh.dev/blog/sticky-active-smooth-responsive-toc and plugged it in.

The existing (blue background) ToC is kept and used instead of the floating one if the screen size is small and the aside wouldn't fit.

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

**Comment by andreialecu** [comment]
There are currently some problems with the Migration page when resizing the window. Looking into it.

File: packages/gatsby/src/components/navigation.js [hunk 1]
```diff
@@ -83,9 +83,7 @@ const MenuEntry = styled(Link)`
 `;
 
 const Content = styled.div`
-  & > * {
-    overflow: auto;
```
Comment (andreialecu) [inline packages/gatsby/src/components/navigation.js:87]: It's not clear what this was for. I didn't notice anything wrong.

It's required in order to get the `position: sticky` to work on the ToC <aside> element.

**Comment by andreialecu** [comment]
Before: 

image

After:

image

**Comment by arcanis** [comment]
Awesome improvement - thanks ! 🎆

**Pull request merged**